packages/api/src/graphql/mutation.private.ts

```diff
@@ -1152,8 +1155,43 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
           rejectionReason
         })
       }
+    },
+
+    // Settings
+    // ==========
+
+    updateSetting: {
+      type: GraphQLSetting,
+      args: {
+        id: {type: GraphQLNonNull(GraphQLString)},
+        input: {type: GraphQLNonNull(GraphQLSettingInput)}
+      },
+
+      async resolve(root, {id, input}, {authenticate, dbAdapter, loaders}) {
+        const fullSetting = await loaders.settingsByID.load(id)
+        const restrictions = fullSetting?.settingRestriction
+
+        const {roles} = authenticate()
+        authorise(CanUpdateSetting, roles)
+
+        if (
```

Comment:
Can we break this `if` statement down into smaller steps/have variables for it? It's hard to follow inline like this.

I agree with @antkiewiczk we should start using a library like `joi` or `yup`